src/index.js

```diff
@@ -485,7 +489,7 @@ const members = {
   },
   assignedSlot: {
     get () {
-      return nodeToSlotMap.get(this) || null;
+      return (nodeToSlotMap.get(this) && hostToModeMap.get(rootToHostMap.get(slotToRootMap.get(nodeToSlotMap.get(this)))) === 'open') ? nodeToSlotMap.get(this) : null;
```

Comment:
I am not sure how expensive the lookups are - potentially splitting it into `const` means that the second half of `&&` is always evaluated unless we do early returns instead.